backend/tracim_backend/lib/core/storage.py

```diff
@@ -0,0 +1,305 @@
+            )
+            raise CannotGetDepotFileDepotCorrupted(
+                "depot file {} is not accessible, depot seems corrupted".format(depot_file.file_id)
+            ) from exc
+
+    def get_raw_file(
+        self,
+        depot_file: UploadedFile,
+        filename: str,
+        default_filename: str,
+        force_download: bool = None,
+        last_modified: datetime = None,
+    ) -> HapicFile:
+        file = self._get_depot_file(depot_file)
+        # INFO - G.M - 2019-08-08 - use given filename in all case but none or
+        # "raw", where filename returned will be a custom one.
+        if not filename or filename == "raw":
+            filename = default_filename
+        return HapicFile(
+            file_object=file,
+            mimetype=file.content_type,
+            filename=filename,
+            as_attachment=force_download,
+            content_length=file.content_length,
+            last_modified=last_modified or None,
+        )
+
+    def get_filepath(
+        self,
+        depot_file: UploadedFile,
+        file_extension: str = "",
+        temporary_prefix: str = TEMPORARY_PREFIX,
+    ):
+        depot_stored_file = self._get_depot_file(depot_file)  # type: StoredFile
+        if self.app_config.UPLOADED_FILES__STORAGE__STORAGE_TYPE == DepotFileStorageType.LOCAL.slug:
+            yield from self._get_valid_content_filepath_legacy(depot_stored_file)
+        else:
+            yield from self._get_valid_content_filepath(
+                depot_stored_file, file_extension=file_extension, prefix=temporary_prefix
+            )
+
+    @contextmanager
+    def preview_generator_filepath_context(
+        self, depot_file: UploadedFile, original_file_extension: str
+    ):
+        """
+
+        :param depot_file:
+        :param original_file_extension:
+        :return:
+        """
+        try:
+            yield from self.get_filepath(
+                depot_file, file_extension=original_file_extension,
+            )
+        except UnavailablePreviewType as exc:
+            raise TracimUnavailablePreviewType(
+                "this kind of preview is not available for this file"
+            ) from exc
+        except UnsupportedMimeType as exc:
+            raise UnavailablePreview("No Preview available for this type of file") from exc
+        except CannotGetDepotFileDepotCorrupted as exc:
+            raise UnavailablePreview(
+                "No Preview available, original file seems no available"
+            ) from exc
+        except PreviewGeneratorPassthroughError as exc:
+            # passthrough as this exception is already supported with
+            # specific error code.
+            raise exc
+        except Exception as exc:
+            logger.warning(self, "Unknown Preview_Generator Exception Occured", exc_info=True)
+            raise UnavailablePreview("No preview available") from exc
```

Comment:
As we catch most preview generator here now instead of on controller level, we can not return here fully explicit error about the content (revision, avatar, etc...) that failed. We can re-catch theses at controller level but i don't thing this is a good idea in term of maintaining (create lot of code).
We can return the depot file id here, but i explictly don't do it as i'm not sure that a good idea from a security perspective. If you think this is ok, we can do it, as this will help us for debugging.